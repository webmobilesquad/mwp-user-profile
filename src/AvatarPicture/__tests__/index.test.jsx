import render from 'preact-render-spy';
import { AvatarPicture } from '..';

describe('<AvatarPicture />', () => {
  test('<AvatarPicture /> displays given src and alt', () => {
    expect(
      render(
        <AvatarPicture
          src="https://mywebportal.michelin-solutions.com/msn-avatars/ef0eccf32b62a0c05d98f4b91cfa9a9a.png"
          alt="John Doe"
        />,
      ),
    ).toMatchSnapshot();
  });

  test('<AvatarPicture /> displays given alt and fallback src', () => {
    expect(
      render(
        <AvatarPicture
          alt="John Doe"
        />,
      ),
    ).toMatchSnapshot();
  });

  test('<AvatarPicture /> displays given src and fallback alt', () => {
    expect(
      render(
        <AvatarPicture
          src="https://mywebportal.michelin-solutions.com/msn-avatars/ef0eccf32b62a0c05d98f4b91cfa9a9a.png"
        />,
      ),
    ).toMatchSnapshot();
  });

  test('<AvatarPicture /> displays fallback src and alt', () => {
    expect(
      render(
        <AvatarPicture />,
      ),
    ).toMatchSnapshot();
  });
});
