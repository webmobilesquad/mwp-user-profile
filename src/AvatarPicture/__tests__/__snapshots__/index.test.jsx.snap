// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarPicture /> <AvatarPicture /> displays fallback src and alt 1`] = `
preact-render-spy (1 nodes)
-------
<img
  src="https://mywebportal.michelin-solutions.com/msn-avatars/73c9c100a15a943fc0446f4b5e880d12.png"
  alt="Avatar"
 />

`;

exports[`<AvatarPicture /> <AvatarPicture /> displays given alt and fallback src 1`] = `
preact-render-spy (1 nodes)
-------
<img
  src="https://mywebportal.michelin-solutions.com/msn-avatars/73c9c100a15a943fc0446f4b5e880d12.png"
  alt="John Doe"
 />

`;

exports[`<AvatarPicture /> <AvatarPicture /> displays given src and alt 1`] = `
preact-render-spy (1 nodes)
-------
<img
  src="https://mywebportal.michelin-solutions.com/msn-avatars/ef0eccf32b62a0c05d98f4b91cfa9a9a.png"
  alt="John Doe"
 />

`;

exports[`<AvatarPicture /> <AvatarPicture /> displays given src and fallback alt 1`] = `
preact-render-spy (1 nodes)
-------
<img
  src="https://mywebportal.michelin-solutions.com/msn-avatars/ef0eccf32b62a0c05d98f4b91cfa9a9a.png"
  alt="Avatar"
 />

`;

exports[`<AvatarPicture /> <AvatarPicture /> take more props in account 1`] = `
preact-render-spy (1 nodes)
-------
<img
  class="className"
  src="https://mywebportal.michelin-solutions.com/msn-avatars/73c9c100a15a943fc0446f4b5e880d12.png"
  alt="Avatar"
 />

`;
